### DISCUSSION TITLE
Force user to re-enter credentials after optional configurable timeout

### DISCUSSION THREAD

**Discussion by lweniger** [belowaverage-org/SuperLauncher#14] [Ideas]
First, let me thank you for writing this great utility! I use it daily.

One feature I would like to see is the option to have a configurable timeout of the supplied credentials when running in Run-As mode. I use this utility for it's "sudo in the system tray" functionality which allows me to stay logged in with non-priviledged credentials while maintaing the convenience of being logged in with admin credentials. The security side of me thinks this is probably too convenient and adding an optional configurable timeout couldn't hurt.

Keep up the great work!

**Comment by krisdb2009** [comment]
This is an interesting idea. I will create a matching feature request.

#15

It might be a bit challenging to implement since the Super Launcher process is restarted any time the user-context is changed.
But it could be possible to just prompt for the password after a timeout and keep Super Launcher running under the user-context the entire time.
